core/src/main/java/com/datastrato/gravitino/storage/relational/service/CatalogMetaService.java

```diff
@@ -55,6 +55,14 @@ public CatalogPO getCatalogPOByMetalakeIdAndName(Long metalakeId, String catalog
     return catalogPO;
   }
 
+  public CatalogPO getCatalogPOById(Long catalogId) {
+    CatalogPO catalogPO =
+        SessionUtils.getWithoutCommit(
+            CatalogMetaMapper.class, mapper -> mapper.selectCatalogMetaById(catalogId));
```

Comment:
What if the catalog and below entity is not found, which exception will it throw?

Reply:
Catalog may be deleted after one role contains the catalog. So I don't throw an exception here.